fairlearn/metrics/true_positive_rate.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+
+from . import MetricsResult
+
+
+def true_positive_rate(y_actual, y_predict, group_id):
+    result = MetricsResult()
+
+    groups = np.unique(group_id)
```

Comment:
This is another scan through one of the arrays. For very large data, it might be more efficient to do the necessary 'contains_key' calls in the main loop, rather than using this to pre-initialise to zero.

Reply:
Per the above.... let's not worry about data that big for now.